src/Symfony/Component/HttpFoundation/BinaryFileResponse.php

```diff
@@ -141,7 +141,7 @@ public function setAutoLastModified()
      */
     public function setAutoEtag()
     {
-        $this->setEtag(sha1_file($this->file->getPathname()));
+        $this->setEtag(substr(base64_encode(hash_file('sha256', $this->file->getPathname(), true)), 0, 32));
```

Comment:
Maybe you wanted to be BC in length? But sha1 is 40 hex characters. Not 32 like md5. So it reduces the length.